incubator/beast/templates/configmap.yaml

```diff
@@ -5,33 +5,46 @@ metadata:
   KAFKA_CONSUMER_MAX_POLL_RECORDS: "{{ .Values.configArgs.kafkaConsumerMaxPollRecords }}"
   KAFKA_CONSUMER_METADATA_MAX_AGE_MS: "{{ .Values.configArgs.kafkaConsumerMetadataMaxAgeMs }}"
   KAFKA_CONSUMER_SESSION_TIMEOUT_MS: "{{ .Values.configArgs.kafkaConsumerSessionTimeoutMs }}"
   KAFKA_CONSUMER_VALUE_DESERIALIZER: "org.apache.kafka.common.serialization.ByteArrayDeserializer"
   KAFKA_TOPIC: "{{ .Values.configArgs.kafkaTopic }}"
-  LOG_LEVEL: "{{ .Values.configArgs.logLevel }}"
+  KAFKA_CONSUMER_AUTO_OFFSET_RESET: "{{ .Values.configArgs.kafkaConsumerAutoOffsetReset }}"
   PROTO_COLUMN_MAPPING: '{{ .Values.configArgs.protoColumnMapping }}'
   PROTO_SCHEMA: "{{ .Values.configArgs.protoSchema }}"
+  CONSUMER_POLL_TIMEOUT_MS: "{{ .Values.configArgs.consumerPollTimeoutMs }}"
+  EXPONENTIAL_BACKOFF_INITIAL_BACKOFF_IN_MS: "{{ .Values.configArgs.exponentialBackoffInitialBackoffInMs }}"
+  EXPONENTIAL_BACKOFF_MAXIMUM_BACKOFF_IN_MS: "{{ .Values.configArgs.exponentialBackoffMaximumBackoffInMs }}"
   READ_QUEUE_CAPACITY: "{{ .Values.configArgs.readQueueCapacity }}"
   COMMIT_QUEUE_CAPACITY: "{{ .Values.configArgs.commitQueueCapacity }}"
-  STATSD_ENABLED: "true"
-  STATSD_HOST: "localhost"
-  STATSD_PORT: "8125"
-  STATSD_PREFIX: "{{ .Values.configArgs.statsdPrefix }}"
-  STENCIL_URL: "{{ .Values.configArgs.stencilURL }}"
-  SENTRY_DSN: "{{ .Values.configArgs.sentryDsn | default " " }}"
-  SENTRY_SAMPLE_RATE: {{ .Values.configArgs.sentrySampleRate | default 1.0 | quote }}
-  OFFSET_ACK_TIMEOUT: {{ .Values.configArgs.offsetAckTimeout | default 60000 | quote }}
-  BQ_WORKER_POLL_TIMEOUT_MS: {{ .Values.configArgs.bqWorkerPollTimeout | default 5000 | quote }}
-  {{ if .Values.configArgs.enableAutoSchemaUpdate }}
-  PROTO_COLUMN_MAPPING_URL: {{ .Values.configArgs.protoColumnMappingURL | default nil }}
-  UPDATE_TABLE_SCHEMA_URL: {{ .Values.configArgs.updateTableSchemaURL }}
+  LOG_LEVEL: "{{ .Values.configArgs.logLevel }}"
+  ENABLE_GCS_ERROR_SINK: "{{ .Values.configArgs.enableGcsErrorSink }}"
+  {{ if .Values.configArgs.enableGcsErrorSink }}
+  GCS_BUCKET: "{{ .Values.configArgs.gcsBucket }}"
+  GCS_PATH_PREFIX: "{{ .Values.configArgs.gcsPathPrefix }}"
+  GCS_WRITER_PROJECT_NAME: "{{ .Values.configArgs.gcsWriterProjectName }}"
   {{ end }}
-  ENABLE_AUTO_SCHEMA_UPDATE: "{{ .Values.configArgs.enableAutoSchemaUpdate | default false }}"
+  STATSD_ENABLED: "{{ .Values.configArgs.statsdEnabled }}"
+  STATSD_HOST: "{{ .Values.configArgs.statsdHost }}"
+  STATSD_PORT: "{{ .Values.configArgs.statsdPort }}"
+  STATSD_PREFIX: "{{ .Values.configArgs.statsdPrefix }}"
+  STENCIL_URL: "{{ .Values.configArgs.stencilUrl }}"
+  SENTRY_DSN: "{{ .Values.configArgs.sentryDsn }}"
+  SENTRY_SAMPLE_RATE: "{{ .Values.configArgs.sentrySampleRate | default 1.0 }}"
+  OFFSET_ACK_TIMEOUT: "{{ .Values.configArgs.offsetAckTimeout | default 60000 }}"
+  GOOGLE_CREDENTIALS: "{{ .Values.configArgs.googleCredentials }}"
+  DEPLOYMENT_NAME: "{{ .Values.configArgs.deploymentName }}"
```

Comment:
you can use `beast.fullname` as deployment name, also make `beast.fullname` as required